### PR TITLE
(GH-373) Refactor PDKCommands to features

### PR DESCRIPTION
**Pull request by jpogran**
This converts all of the PDK commands to use the Feature infrastructure.

### CODE REVIEW

**Comment by jpogran** [comment]
@glennsarti I was at first opposed to your idea, one class with ~5 (with more to come) methods didn't seem like a better idea than several classes with one method, but I tried it out and I like it better. If it gets too unwieldy we can extract methods and classes from it later. Will have a PR up soon

**Comment by jpogran** [comment]
@glennsarti updated

File: src/feature/PdkCommandFeature.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    [
+      PDKCommandStrings.PdkNewModuleCommandId,
+      PDKCommandStrings.PdkNewClassCommandId,
+      PDKCommandStrings.PdkNewTaskCommandId,
+      PDKCommandStrings.PdkTestUnitCommandId,
+      PDKCommandStrings.PdkValidateCommandId
+    ].forEach((id)=>{
+      context.subscriptions.push(vscode.commands.registerCommand(id, () => {
+        this.pdkNewModule();
```
Comment (glennsarti) [inline src/feature/PdkCommandFeature.ts]: Not sure if this was your intent.  Every command will call this.pdkNewModule();  I assume what you really mean is

``` typescript
context.subscriptions.push(vscode.commands.registerCommand(PDKCommandStrings.PdkNewModuleCommandId, () => {
        this.pdkNewModule();
}
this.logger.debug("Registered " + PDKCommandStrings.PdkNewModuleCommandId+ " command");

context.subscriptions.push(vscode.commands.registerCommand(PDKCommandStrings.PdkNewClassCommandId, () => {
        this.pdkNewClass();
}
this.logger.debug("Registered " + PDKCommandStrings.PdkNewClassCommandId+ " command");

...
```

File: src/feature/PdkCommandFeature.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      }));
+      this.logger.debug("Registered " + id + " command");
+    });
+  }
+
+  public dispose(): any {
+    this.terminal.dispose();
+  }
+
+  public pdkNewModule() {
```
Comment (glennsarti) [inline src/feature/PdkCommandFeature.ts]: nit: these could probably all be private.  Not sure.

**Comment by jpogran** [comment]
Superseded by #389

**Pull request closed without merging**